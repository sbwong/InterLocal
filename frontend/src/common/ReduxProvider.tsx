import React, { ReactNode } from "react";

import { Provider } from "react-redux";
import { store } from "../app/store";

export interface ReduxProviderProps {
	children: ReactNode;
}

export default function ReduxProvider(props: ReduxProviderProps) {
	return <Provider store={store}>{props.children}</Provider>;
}
