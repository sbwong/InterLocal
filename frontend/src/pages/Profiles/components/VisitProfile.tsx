import Link from "@material-ui/core/Link";
import React from "react";
import VerifiedUserIcon from "@material-ui/icons/VerifiedUser";

interface VisitProfileProps {
	user_id: number;
	username: string;
	is_admin: boolean;
}

export default function VisitProfile({
	user_id,
	username,
	is_admin,
}: VisitProfileProps) {
	const linkPath: string = "/Profile/" + user_id;

	return username ? (
		<Link href={linkPath}>
			{username}
			{is_admin && <VerifiedUserIcon fontSize={"inherit"} />}
		</Link>
	) : null;
}
