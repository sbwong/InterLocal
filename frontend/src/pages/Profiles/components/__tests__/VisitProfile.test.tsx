import React from "react";
import ReactDOM from "react-dom";
import ReduxProvider from "../../../../common/ReduxProvider";
import VisitProfile from "../VisitProfile";
import { render } from "@testing-library/react";

it("renders without crashing", () => {
	const div = document.createElement("div");

	ReactDOM.render(
		<ReduxProvider>
			<VisitProfile username={"rochouhan6"} user_id={131} />
		</ReduxProvider>,
		div
	);
});

it("renders VisitProfile page", () => {
	const { visitprofile } = render(
		<ReduxProvider>
			<VisitProfile username={"rochouhan6"} user_id={131} />
		</ReduxProvider>
	);
	expect(visitprofile).toMatchSnapshot();
});
