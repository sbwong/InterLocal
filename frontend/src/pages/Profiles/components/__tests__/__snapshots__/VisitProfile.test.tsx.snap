// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders VisitProfile page 1`] = `undefined`;
